src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -154,10 +154,10 @@ bool ChaseMovementGenerator<T>::DoUpdate(T* owner, uint32 time_diff)
     i_leashExtensionTimer.Update(time_diff);
     if (i_leashExtensionTimer.Passed())
     {
-        i_leashExtensionTimer.Reset(5000);
+        i_leashExtensionTimer.Reset(1500);
         if (owner->HasUnitState(UNIT_STATE_CHASE) && owner->movespline->Finalized() && !target->isMoving())
```

Comment:
Why do we have this `target->isMoving()` check?

Originally, in vMangos, this timer only ticks when owner->movespline->Finalized() is true (without checking owner->HasUnitState(UNIT_STATE_CHASE)). Otherwise, it resets the timer.

Reply:
If @PkllonG would like to chime in and provide the reasoning, we can add it back, but for now I've adjusted the checks for parity with VMaNGOS.

Reply:
It was added because, I think, it is forbidden to work when the player is running away. Otherwise, the player cannot leave the battle.

Reply:
If there's a better way, feel free to change or delete it

Reply:
The system as I understand it should not rely on the target's movement hardly at all. If the creature stands still, then it should reset its leash. The `i_recalculateTravel` (I believe) indicates the mob is fanning/spreading out in front of the player, so that also resets the movement leash timer check.

I'm also not convinced about the reset interval either used in this PR or as used by VMaNGOS, but most videos I've seen describing the system suggest it is 1.5s, so that's what I've chosen to replace it with here, as it does seem to help mitigate premature leashing issues.